### PR TITLE
AP_HAL: Split the AP_HAL_Boards.h header into per board sub headers

### PR DESCRIPTION
**Pull request by kwikius**
Makes it easier to find the bit you want.

### CODE REVIEW

File: libraries/AP_HAL/board/vrbrain.h [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#endif
+#define HAL_GPIO_A_LED_PIN        25
+#define HAL_GPIO_B_LED_PIN        26
+#define HAL_GPIO_C_LED_PIN        27
+#define HAL_GPIO_LED_ON           HIGH
+#define HAL_GPIO_LED_OFF          LOW
+#define EXTERNAL_LED_GPS          28    // GPS LED - AN10
+#define EXTERNAL_LED_ARMED        29    // Armed LED - AN11
+#define EXTERNAL_LED_MOTOR1       30    // Motor1 LED - AN8
+#define EXTERNAL_LED_MOTOR2       31    // Motor2 LED - AN12
```
Comment (khancyr) [inline libraries/AP_HAL/board/vrbrain.h]: miss new line at file end

File: libraries/AP_HAL/board/sitl.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+#define HAL_BOARD_TERRAIN_DIRECTORY "terrain"
+#define HAL_PARAM_DEFAULTS_PATH "etc/defaults.parm"
+#define HAL_INS_DEFAULT HAL_INS_HIL
+#define HAL_BARO_DEFAULT HAL_BARO_HIL
+#define HAL_COMPASS_DEFAULT HAL_COMPASS_HIL
+#define HAL_GPIO_A_LED_PIN        61
+#define HAL_GPIO_B_LED_PIN        48
+#define HAL_GPIO_C_LED_PIN        117
+#define HAL_GPIO_LED_ON           LOW
+#define HAL_GPIO_LED_OFF          HIGH
```
Comment (khancyr) [inline libraries/AP_HAL/board/sitl.h]: miss new line at file end

**Comment by khancyr** [comment]
Like the idea ! But it wil finish by pollute the directory no? Maybe those files should be put in board subdirectory, what do you think ?

**Comment by kwikius** [comment]
@khancyr.  Look again. I did put them in a subdirectory. :)
I fixed the line endings I hope.

**Comment by magicrub** [comment]
This will probably need a wiki update, but otherwise LGTM. @lucasdemarchi , assigning to you

**Comment by kwikius** [comment]
@magicrub  Great.  With the header a bit more manageable , it should also be obvious that the sensors defines could be moved out to their respective libraries AP_Baro, AP_Compass, AP_InertialSensor In fact boards are dependent on sensors rather than vice versa. Then (for example)
https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Compass/AP_Compass_AK8963.cpp#L22
this type of logic would be inverted. The board would include its sensor headers not vice versa, which would speed compilation since only required sensors would be included , but one step at a time

**Comment by lucasdemarchi** [comment]
@kwikius can you squash the 2 commits?

**Comment by magicrub** [comment]
@lucasdemarchi if it's just the squash holding it up, I can do that and merge for you.

**Comment by kwikius** [comment]
@lucasdemarchi @magicrub Whichever way you prefer. I did a v2 branch https://github.com/kwikius/ardupilot/tree/split_ap_hal_boards_v2.

**Comment by lucasdemarchi** [comment]
Applied, thanks.

**Pull request closed without merging**